src/managers/GitRefManager.js

```diff
@@ -103,6 +103,19 @@ export class GitRefManager {
     const normalizeValue = value => value.trim() + '\n'
     await fs.write(join(gitdir, ref), normalizeValue(value), 'utf8')
   }
+  static async writeSymbolicRef ({ fs: _fs, gitdir, ref, value }) {
+    const fs = new FileSystem(_fs)
+    // TODO: Validate input
+    if (!value.startsWith('refs/')) {
+      throw new GitError(E.NotAnOidFail, { value })
```

Comment:
This is the wrong kind of error. Either skip this check or use an E.InternalFail probably.